crypto/x509/x509_vfy.c

```diff
@@ -448,8 +448,8 @@ static int check_chain(X509_STORE_CTX *ctx)
 
     /*-
      *  must_be_ca can have 1 of 3 values:
-     * -1: we accept both CA and non-CA certificates, to allow direct
-     *     use of self-signed certificates (which are marked as CA).
+     * -1: we accept both CA and non-CA certificates, to allow direct use
+     *     of self-signed certificates (which may be marked as CA or not).
      * 0:  we only accept non-CA certificates.  This is currently not
      *     used, but the possibility is present for future extensions.
```

Comment:
Actually this seems to be used for proxy certs.

Reply:
I've corrected that old comment accordingly.